packages/jest-circus/src/utils.ts

```diff
@@ -159,14 +159,15 @@ function checkIsError(error: unknown): error is Error {
 }
 
 export const callAsyncCircusFn = (
-  fn: Circus.AsyncFn,
+  testOrHook: Circus.TestEntry | Circus.Hook,
   testContext: Circus.TestContext | undefined,
-  asyncError: Circus.Exception,
   {isHook, timeout}: {isHook?: boolean | null; timeout: number},
 ): Promise<unknown> => {
   let timeoutID: NodeJS.Timeout;
   let completed = false;
 
+  const {fn, asyncError} = testOrHook;
+
   return new Promise((resolve, reject) => {
     timeoutID = setTimeout(
       () => reject(_makeTimeoutMessage(timeout, !!isHook)),
```

Comment:
```suggestion
      () => reject(_makeTimeoutMessage(timeout, isHook)),
```

after the change above